src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrScanner.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Darc.Models.VirtualMonoRepo;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.Extensions.Logging;
+
+
+namespace Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+/// <summary>
+/// Class that scans the VMR for cloaked files that shouldn't be inside.
+/// </summary>
+public class VmrScanner : IVmrScanner
+{
+    private const string _vmrPreserveAttribute = "vmr-preserve";
+
+    private readonly IVmrDependencyTracker _dependencyTracker;
+    private readonly IProcessManager _processManager;
+    private readonly IVmrInfo _vmrInfo;
+    private readonly ILogger<VmrScanner> _logger;
+
+    public VmrScanner(
+        IVmrDependencyTracker dependencyTracker,
+        IProcessManager processManager,
+        IVmrInfo vmrInfo,
+        ILogger<VmrScanner> logger)
+    {
+        _dependencyTracker = dependencyTracker;
+        _processManager = processManager;
+        _vmrInfo = vmrInfo;
+        _logger = logger;
+    }
+
+    public async Task<List<string>> ListCloakedFiles(CancellationToken cancellationToken)
+    {
+        await _dependencyTracker
+            .InitializeSourceMappings(_vmrInfo.VmrPath / VmrInfo.SourcesDir / VmrInfo.SourceMappingsFileName);
+
+        var cloakedFilesList = new List<string>();
```

Comment:
nit: I wouldn't put the hungarian-notation-like "List" suffix there. It's a list already

Reply:
Removed. Didn't know that was called the hungarian notation